plsync/mlabconfig.py

```diff
@@ -45,13 +64,152 @@ def parse_flags():
+def export_server_records_v6(output, sites, decoration=''):
+  comment(output, 'hosts v6%s' % (' decorated' if decoration else ''))
+  for site in sites:
+      # TODO: change site['nodes'] to a pre-sorted list type.
+      for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+          if node.ipv6_is_enabled():
+              write_aaaa_record(
+                  output, node.recordname(decoration), node.ipv6())
+
+
+def export_experiment_records(output, sites, experiments):
+  for experiment in experiments:
+    if experiment['index'] is None:
+      # Ignore experiments without an IP address.
+      continue
+    export_experiment_records_v4(output, sites, experiment)
+    export_experiment_records_v4(output, sites, experiment, decoration='v4')
+
+    export_experiment_records_v6(output, sites, experiment)
+    export_experiment_records_v6(output, sites, experiment, decoration='v6')
+
+
+def export_experiment_records_v4(output, sites, experiment, decoration=''):
+  comment(output, '%s v4%s' % (experiment.dnsname(), (
+      ' decorated' if decoration else '')))
+  for site in sites:
+    # TODO: change site['nodes'] to a pre-sorted list type.
+    for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+        # TODO: should sitenames exclude mlab4's?
+        write_a_record(output, experiment.sitename(node, decoration),
+                       experiment.ipv4(node))
+        write_a_record(output, experiment.recordname(node, decoration),
+                       experiment.ipv4(node))
+
+
+def export_experiment_records_v6(output, sites, experiment, decoration=''):
+  comment(output, '%s v6%s' % (experiment.dnsname(), (
+      ' decorated' if decoration else '')))
+  for site in sites:
+    # TODO: change site['nodes'] to a pre-sorted list type.
+    for node in sorted(site['nodes'].values(), key=lambda n: n.hostname()):
+      # TODO: should sitenames exclude mlab4's?
+      if (node.ipv6_is_enabled() and experiment.ipv6(node)):
+        write_aaaa_record(output, experiment.sitename(node, decoration),
+                          experiment.ipv6(node))
+        write_aaaa_record(output, experiment.recordname(node, decoration),
+                          experiment.ipv6(node))
+
+
+def export_mlab_zone_records(output, sites, experiments):
+    export_router_and_switch_records(output, sites)
+    export_pcu_records(output, sites)
+    export_server_records(output, sites)
+    export_experiment_records(output, sites, experiments)
+
+
+def serial_rfc1912(ts):
+    """Returns an rfc1912 style serial id (YYYYMMDDnn) for a DNS zone file."""
+    # RFC1912 (http://www.ietf.org/rfc/rfc1912.txt) recommends 'nn' as the
+    # revision. However, identifying and incrementing this value is a manual,
+    # error prone step. Instead, the following calculates 'nn' from HH:MM (00:00
+    # to 23:59, or 0 to 1439) by calculating the corresponding value in the
+    # range 00 to 99. 'nn' increases by 1 about every 15 minutes.
```

Comment:
I agree that rerunning the generation within 15 min is probably more likely than 100 times in 24 hours.

My impression is that the date form is preferred by convention. You're right that it's small enough to fit and be unique every second. IIRC, switching to timestamps would require some careful planning so that the serial is incremented twice in such a way to overflow the 32-bit value it's stored in. Otherwise just switching over would make the current timestamp (1475......) appear earlier than the current serial (2016......).

Could we defer this design question to a future change?